.github/workflows/linux/build.bash

```diff
@@ -0,0 +1,85 @@
+set -exo pipefail
+
+export CCACHE_DIR=/icinga2/ccache
+CMAKE_OPTS=''
+
+case "$DISTRO" in
+  amazonlinux:*)
+    amazon-linux-extras install -y epel
+    yum install -y bison ccache cmake gcc-c++ flex ninja-build \
+      {libedit,mariadb,ncurses,openssl,postgresql,systemd}-devel
+
+    yum install -y bzip2 tar wget
+    wget https://boostorg.jfrog.io/artifactory/main/release/1.69.0/source/boost_1_69_0.tar.bz2
+    tar -xjf boost_1_69_0.tar.bz2
+
+    (
+      cd boost_1_69_0
+      export PATH="/usr/lib64/ccache:$PATH"
+      ./bootstrap.sh --with-libraries=context,coroutine,date_time,filesystem,iostreams,program_options,regex,system,thread
+      ./b2 link=static
+    )
+
+    CMAKE_OPTS='-DBOOST_INCLUDEDIR=/boost_1_69_0 -DBOOST_LIBRARYDIR=/boost_1_69_0/stage/lib'
+    ;;
+
+  centos:*)
+    yum install -y centos-release-scl epel-release
+    yum install -y bison ccache cmake devtoolset-11-gcc-c++ flex ninja-build \
+      {boost169,libedit,mariadb,ncurses,openssl,postgresql,systemd}-devel
+
+    export PATH="/opt/rh/devtoolset-11/root/usr/bin:$PATH"
+    CMAKE_OPTS='-DBOOST_INCLUDEDIR=/usr/include/boost169 -DBOOST_LIBRARYDIR=/usr/lib64/boost169'
+    ;;
+
+  debian:*|ubuntu:*)
+    apt-get update
+    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-{recommends,suggests} -y bison \
+      ccache cmake flex g++ lib{boost-all,edit,mariadb,ncurses,pq,ssl,systemd}-dev ninja-build
+    ;;
+
+  fedora:*)
+    dnf install -y bison ccache cmake flex gcc-c++ ninja-build \
+      {boost,libedit,mariadb,ncurses,openssl,postgresql,systemd}-devel
+    ;;
+
+  opensuse/*)
+    zypper in -y bison ccache cmake flex gcc-c++ ninja {lib{edit,mariadb,openssl},ncurses,postgresql,systemd}-devel \
+      libboost_{context,coroutine,filesystem,iostreams,program_options,regex,system,test,thread}-devel
+    ;;
+
+  rockylinux:*)
+    dnf install -y 'dnf-command(config-manager)' epel-release
+
+    case "$DISTRO" in
+      *:8)
+        dnf config-manager --enable powertools
+        ;;
+      *)
+        dnf config-manager --enable crb
+        ;;
+    esac
+
+    dnf install -y bison ccache cmake gcc-c++ flex ninja-build \
+      {boost,libedit,mariadb,ncurses,openssl,postgresql,systemd}-devel
+    ;;
+esac
+
+mkdir /icinga2/build
+cd /icinga2/build
+
+cmake \
+  -GNinja \
+  -DCMAKE_C_COMPILER=/icinga2/.github/workflows/linux/gcc \
+  -DCMAKE_CXX_COMPILER=/icinga2/.github/workflows/linux/g++ \
```

Comment:
You shouldn't have to provide custom wrappers here, have a look at `/usr/lib/ccache/` and `/usr/lib64/ccache/`.

Reply:
Show me that working on CentOS 7 and I'll change it in an instant.

Reply:
```console
$ docker run --rm -it centos:7
[root@e170f12a413d /]# yum install epel-release
[...]
Complete!
[root@e170f12a413d /]# yum install ccache gcc-c++
[...]
Complete!
[root@e170f12a413d /]# echo 'int answer() { return 42; }' > test.cpp
[root@e170f12a413d /]# /usr/lib64/ccache/g++ -c test.cpp 
[root@e170f12a413d /]# /usr/lib64/ccache/g++ -c test.cpp 
[root@e170f12a413d /]# ccache -s
cache directory                     /root/.ccache
primary config                      /root/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
stats updated                       Fri Jan 27 11:51:24 2023
cache hit (direct)                     1
cache hit (preprocessed)               0
cache miss                             1
cache hit rate                     50.00 %
cleanups performed                     0
files in cache                         2
cache size                           8.2 kB
max cache size                       5.0 GB
```

Cache hits > 0, so seems to just work?

Reply:
No, no, no! Not your random hello world program. Icinga 2.

Reply:
Why should this make a difference. If you're claiming that this doesn't work, please back up this claim.

Reply:
You're using gcc-c++. I use devtoolset-11-gcc-c++. That's the difference.

```
[root@ede6466ca7b9 /]# ls /usr/lib64/ccache
[root@ede6466ca7b9 /]#
```

Reply:
It would have been so much easier for both of us if you could just have said that at the beginning.

Reply:
Adding the symlink yourself would also do the trick, ccache also correctly uses the next binary from `$PATH`:

```console
[root@1416bc35fc5c /]# ln -s /usr/bin/ccache /usr/lib64/ccache/g++
[root@1416bc35fc5c /]# export PATH=/usr/lib64/ccache:/opt/rh/devtoolset-11/root/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
[root@1416bc35fc5c /]# g++ --version
g++ (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```